framework/graphics/dx12_dump_resources.cpp

```diff
@@ -0,0 +1,365 @@
+/*
+** Copyright (c) 2023 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "graphics/dx12_dump_resources.h"
+// TODO: It should change the file name of "vulkan"
+#include "generated/generated_vulkan_struct_to_json.h"
+#include "util/platform.h"
+#include "util/logging.h"
+#include "util/image_writer.h"
+#include "graphics/dx12_util.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(graphics)
+
+// TEST_READABLE is only for test because data type could be various.
+// But here uses fixed type.
+// float for vertex, index, constant buffer.
+// image for shader resource, render target.
+const bool TEST_READABLE = false;
```

Comment:
To output readable data. Set TEST_READABLE = true.
Float for vertex, index, constant buffer.
Image for shader resource, render target.